src/chain/chain.rs

```diff
@@ -798,20 +811,19 @@ impl Chain {
         self.state.our_info().member_nodes()
     }
 
-    fn elders_and_adults(&self) -> impl Iterator<Item = &PublicId> {
+    /// Returns joined adults and elders from our section.
```

Comment:
A node can leave the section and then rejoin which would maintain its original age (although it's not fully implemented yet). To support this, we keep the info about the node, just mark it as `Left`. So yes, there can be nodes that are "not joined".

Reply:
Also, there is a third state - `Relocating`. That is for nodes that's already been decided to be relocated, but have not relocate yet for various reasons. A `Relocating` node is not considered as "joined".

Reply:
Hm. But `joined` means it _has_ at some point joined. It doesn't mean that it "is" joined (since there is no such thing as "being" joined), and that it leaves doesn't mean that it didn't join, so "not joined" is a grammatical hick-up. The wording is unfortunate.
We're describing a **current** state with a mix of _past tense_ and _present tense_.
`Joined` is past tense, `Left` is past tense. `Relocating` is present.
For describing current state (which this does), it would be better to stay consistent with that.

Suggestion for a more consistent and straight forward language (no past tense used):
`Member` replaces `Joined`.
`Away` replaces `Left`. `Away` describes a current state, and would indicate that we consider the possibility that it will come back - more so than `Left` I would say. It is not a `Member` while `Away`, per definition.
`Relocating` would stay as is. It also per definition means it is not a `Member`.

Reply:
Those are good suggestions, but I'd probably make them in a different PR as this one has different focus.